ui/pages/confirmations/components/confirm/info/info.tsx

```diff
@@ -14,17 +15,24 @@ const ConfirmationInfoConponentMap = {
 
 type ConfirmationType = keyof typeof ConfirmationInfoConponentMap;
 
-const Info: React.FC = memo(() => {
+const Info: React.FC = () => {
   const currentConfirmation = useSelector(currentConfirmationSelector);
 
   if (!currentConfirmation?.type) {
     return null;
   }
 
-  const InfoComponent =
+  let InfoComponent =
     ConfirmationInfoConponentMap[currentConfirmation?.type as ConfirmationType];
 
+  if (currentConfirmation.type === TransactionType.signTypedData) {
+    const { version } = currentConfirmation?.msgParams ?? {};
+    if (version === 'V1') {
+      InfoComponent = TypedSignV1Info;
+    }
+  }
```

Comment:
Should this be abstracted in the ConfirmationInfoConponentMap?
We could isolate Confirmation component selection logic within it (in my opinion making it more isolate and readable). Something like:
```js
const getTypedSignInfo = (currentConfirmation) => {
  const { version } = currentConfirmation?.msgParams ?? {};
  if (version === 'V1') {
    return TypedSignV1Info;
  }

  return TypedSignInfo;
}

const ConfirmationInfoConponentMap = {
  [TransactionType.personalSign]: () => PersonalSignInfo,
  [TransactionType.signTypedData]: getTypedSignInfo,
};

...

const InfoComponent =
    ConfirmationInfoConponentMap[currentConfirmation?.type as ConfirmationType](currentConfirmation);
```

What do you think?

(Also just noticed, ConfirmationInfoConponentMap has a typo.. should be ConfirmationInfoComponentMap)

Reply:
PR is updated